wallet-client/Run.hs

```diff
@@ -0,0 +1,25 @@
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Run
+  ( runClient
+  ) where
+
+import           Data.Functor.Contravariant (contramap)
+import           Data.Text (Text, pack)
+
+import           Cardano.BM.Data.Tracer (ToLogObject (..))
+import           Cardano.BM.Trace (Trace, appendName)
+import           Ouroboros.Consensus.NodeId (CoreNodeId (..))
+
+import           CLI
+import           WalletClient
+
+runClient :: CLI -> Trace IO Text -> IO ()
+runClient CLI{..} tracer = do
+    let CoreNodeId nid = cliCoreNodeId
+    tracer'
+      <- contramap pack . toLogObject <$>
+          appendName ("Wallet " <> pack (show nid)) tracer
```

Comment:
Not an issue with this PR but I can't help but point it out: `appendName` has a bad type. It should be `LoggerName -> Trace m a -> Trace m a`, but they put it into `m` instead, for no apparent reason.

Reply:
^ @CodiePP

Reply:
thanks for pointing this out. We will have to check this, as this has been annoying at other places too. Probably, it is just legacy we carry around..

Reply:
Anything that can be a pure function *should* be a pure function.

Reply:
This is fixed now.